### PR TITLE
Use build-appimage.sh from Brander

### PR DESCRIPTION
**Pull request by FlexW**
Signed-off-by: Felix Weilbach <felix.weilbach@nextcloud.com>

### CODE REVIEW

File: admin/linux/build-appimage.sh [hunk 1]
```diff
@@ -2,71 +2,82 @@
...
-export SUFFIX=${DRONE_PULL_REQUEST:=master}
-if [ $SUFFIX != "master" ]; then
-    SUFFIX="PR-$SUFFIX"
+# Set defaults
+if [ "$BUILD_UPDATER" != "OFF" ]; then
+    BUILD_UPDATER=ON
 fi
 
-#QtKeyChain v0.10.0
+#QtKeyChain master
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:27]: let's put v0.10.0 in the comment to match what we fetch

File: admin/linux/build-appimage.sh [hunk 2]
```diff
@@ -2,71 +2,82 @@
...
 git clone https://github.com/frankosterfeld/qtkeychain.git
 cd qtkeychain
 git checkout v0.10.0
 mkdir build
 cd build
 cmake -D CMAKE_INSTALL_PREFIX=/usr ../
-make -j4
-make install
+make -j$(nproc)
+make -j$(nproc) DESTDIR=/app install
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:36]: remove since it is not needed

File: admin/linux/build-appimage.sh [hunk 3]
```diff
@@ -2,71 +2,82 @@
...
+    -D NO_SHIBBOLETH=1 \
     -D BUILD_TESTING=OFF \
-    -D BUILD_UPDATER=ON \
-    -DMIRALL_VERSION_SUFFIX=PR-$DRONE_PULL_REQUEST \
-    -DMIRALL_VERSION_BUILD=$DRONE_BUILD_NUMBER \
-    $DRONE_WORKSPACE
-make -j4
-make DESTDIR=/app install
+    -D BUILD_UPDATER=$BUILD_UPDATER \
+    -D QTKEYCHAIN_LIBRARY=/app/usr/lib/x86_64-linux-gnu/libqt5keychain.so \
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:48]: so we install either in `/app/usr` or we install in `/usr` but not both and then we do not know where to look for files

File: admin/linux/build-appimage.sh [hunk 4]
```diff
@@ -2,71 +2,82 @@
...
+    -D QTKEYCHAIN_INCLUDE_DIR=/app/usr/include/qt5keychain/ \
+    -D MIRALL_VERSION_BUILD=$BUILDNR \
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:50]: does not match the need of CI running in Drone
next line will also be wrong

File: admin/linux/build-appimage.sh [hunk 5]
```diff
@@ -2,71 +2,82 @@
...
+    /home/user/
+make -j$(nproc)
+make -j$(nproc) DESTDIR=/app install
 
 # Move stuff around
 cd /app
 
+if [ -d "./usr/lib/x86_64-linux-gnu/${APPNAME}" ]; then
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:58]: why is this needed ?

File: admin/linux/build-appimage.sh [hunk 6]
```diff
@@ -2,71 +2,82 @@
...
+    mv ./usr/lib/x86_64-linux-gnu/${APPNAME}/* ./usr/lib/x86_64-linux-gnu/
+fi
 mv ./usr/lib/x86_64-linux-gnu/* ./usr/lib/
+if [ -d "./usr/lib/${APPNAME}" ]; then
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:62]: same question

File: admin/linux/build-appimage.sh [hunk 7]
```diff
@@ -82,18 +93,19 @@ cp -P -r /usr/lib/x86_64-linux-gnu/nss ./usr/lib/
...
 cd /build
-wget --ca-directory=/etc/ssl/certs/ -c "https://github.com/probonopd/linuxdeployqt/releases/download/continuous/linuxdeployqt-continuous-x86_64.AppImage"
+wget --ca-directory=/etc/ssl/certs -c "https://github.com/probonopd/linuxdeployqt/releases/download/continuous/linuxdeployqt-continuous-x86_64.AppImage"
 chmod a+x linuxdeployqt*.AppImage
 ./linuxdeployqt-continuous-x86_64.AppImage --appimage-extract
 rm ./linuxdeployqt-continuous-x86_64.AppImage
 unset QTDIR; unset QT_PLUGIN_PATH ; unset LD_LIBRARY_PATH
 export LD_LIBRARY_PATH=/app/usr/lib/
-./squashfs-root/AppRun ${DESKTOP_FILE} -bundle-non-qt-libs -qmldir=$DRONE_WORKSPACE/src/gui
+./squashfs-root/AppRun ${DESKTOP_FILE} -bundle-non-qt-libs -qmldir=/home/user/src/gui
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:102]: will not work in drone CI

File: admin/linux/build-appimage.sh [hunk 8]
```diff
@@ -82,18 +93,19 @@ cp -P -r /usr/lib/x86_64-linux-gnu/nss ./usr/lib/
...
 
 # Set origin
-./squashfs-root/usr/bin/patchelf --set-rpath '$ORIGIN/' /app/usr/lib/libnextcloudsync.so.0
+./squashfs-root/usr/bin/patchelf --set-rpath '$ORIGIN/' /app/usr/lib/lib${APPNAME}sync.so.0
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:105]: since yesterday, I really wonder why we keep this line

File: admin/linux/build-appimage.sh [hunk 9]
```diff
@@ -82,18 +93,19 @@ cp -P -r /usr/lib/x86_64-linux-gnu/nss ./usr/lib/
...
 
 # Build AppImage
 ./squashfs-root/AppRun ${DESKTOP_FILE} -appimage
 
-mv Nextcloud*.AppImage Nextcloud-${SUFFIX}-${DRONE_COMMIT}-x86_64.AppImage
+#move AppImage
+mv /build/*.AppImage /home/user/
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:111]: will not work in drone CI

**Review by mgallien — CHANGES_REQUESTED** [review]

File: admin/linux/build-appimage.sh (hunk 2 above)
Comment (FlexW) [reply admin/linux/build-appimage.sh:36]: you mean DESTDIR?

File: admin/linux/build-appimage.sh (hunk 8 above)
Comment (FlexW) [reply admin/linux/build-appimage.sh:105]: me too

File: admin/linux/build-appimage.sh [hunk 10]
```diff
@@ -2,74 +2,79 @@
...
+# QtKeyChain
 git clone https://github.com/frankosterfeld/qtkeychain.git
 cd qtkeychain
 git checkout v0.10.0
 mkdir build
 cd build
-cmake -D CMAKE_INSTALL_PREFIX=/usr ../
-make -j4
-make install
+cmake -D CMAKE_INSTALL_PREFIX=/app/usr ..
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:35]: cannot we use `ninja` here ?

File: admin/linux/build-appimage.sh [hunk 11]
```diff
@@ -2,74 +2,79 @@
...
+make -j$(nproc)
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:36]: better use this syntax
```
cmake --build . --target all
```

File: admin/linux/build-appimage.sh [hunk 12]
```diff
@@ -2,74 +2,79 @@
...
+make -j$(nproc) install
```
Comment (mgallien) [inline admin/linux/build-appimage.sh:37]: better use this syntax
```
cmake --build . --target install
```

**Review by mgallien — COMMENTED** [review]
thanks, see my comments about `cmake` general usage and let's use `cmake` directly even for build (maximize portability and makes it easy to switch generator)

**Review by mgallien — APPROVED** [review]
very nice

**Comment by mgallien** [comment]
> very nice

please clean history (I again missed that before approving)

**Review by allexzander — APPROVED** [review]
